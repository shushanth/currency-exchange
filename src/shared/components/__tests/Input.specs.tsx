import React from 'react';
import renderer from 'react-test-renderer';

import { CEInput } from '../index';
import { mount } from 'enzyme';

describe('<CEInput /> snapshots', () => {
  test('tree snapshots match', () => {
    const inputTree = renderer
      .create(<CEInput val={1} inputType={'number'} />)
      .toJSON();
    expect(inputTree).toMatchSnapshot();
  });
});

describe('Component: <CEInput />', () => {
  let CEInputComponentWrapper: any;
  beforeEach(() => {
    CEInputComponentWrapper = mount(<CEInput val={1} inputType={'number'} />);
  });

  test('it should render properly', () => {
    expect(CEInputComponentWrapper.length).toBeDefined();
  });
  test('it should match the input value with the props passed value', () => {
    expect(CEInputComponentWrapper.prop('val')).toEqual(1);
    expect(CEInputComponentWrapper.prop('inputType')).toEqual('number');
  });
});
