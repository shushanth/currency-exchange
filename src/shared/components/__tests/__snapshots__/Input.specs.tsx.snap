// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CEInput /> snapshots tree snapshots match 1`] = `
<div
  className="sc-AxjAm fzHlOB"
>
  <input
    className="sc-AxirZ drxayQ"
    onChange={[Function]}
    placeholder=""
    type="number"
    value={1}
  />
</div>
`;
